LoopKit/InsulinKit/DoseStore.swift

```diff
@@ -900,12 +899,28 @@ extension DoseStore {
     ///   - start: The date on and after which to include doses
     ///   - completion: A closure called on completion
     ///   - result: The doses along with schedule basal
-    private func getPumpEventDoseEntriesForSavingToHealthStore(lastBasalEndDate: Date, completion: @escaping (_ result: DoseStoreResult<[DoseEntry]>) -> Void) {
+    private func getPumpEventDoseEntriesForSavingToHealthStore(startingAt: Date, completion: @escaping (_ result: DoseStoreResult<[DoseEntry]>) -> Void) {
+        // Can't store to HealthKit if we don't know end of reconciled range, or if we already have doses after the end
+        guard let endingAt = self.lastPumpEventsReconciliation, endingAt > startingAt else {
```

Comment:
[nit] Technically don't need `self.` here now that it is pulled out of the closure.